deck/models.py

```diff
@@ -272,6 +285,26 @@ def proposals(self):
             pk__in=self.activities.values_list('pk', flat=True)
         )
 
+    def has_activities(self):
+        return self.activities.exists()
+
+    def add_proposal_to_slot(self, proposal, slot_index):
+        proposal.track = self
+        proposal.is_approved = True
+        proposal.track_order = slot_index
+        proposal.save()
+
+    def add_activity_to_slot(self, activity, slot_index):
```

Comment:
Just because from a domain language view, is more coherent to think of an track adding a proposal/activity instead of the opposite. This is just a design decision closer to the domain language (see [here](https://www.agilealliance.org/glossary/ubiquitous-language/) for more reference).